### ISSUE TITLE
TSError: ⨯ Unable to compile TypeScript (v3.2.0)

### ISSUE THREAD

**Issue by WhitWaldo** [TypeStrong/ts-node#385]
**ts-node Version:**  3.2.0

**Description:**
I updated my npm packages today (including ts-node) and then attempted to re-run my unit tests (via Karma/Jasmine).  Unfortunately, I'm unable to share my original project with you, but while investigating this issue, I found #156, so I wanted to create a new issue to indicate that it seems to be a recurring problem.

**Actual behavior:**
I received the following error in the console.

> 09 07 2017 17:48:18.989:ERROR [config]: Invalid config file!
>   TSError: ⨯ Unable to compile TypeScript
> Cannot find type definition file for 'jasmine'. (2688)
> Cannot find type definition file for 'node'. (2688)
> Cannot find type definition file for 'sinon'. (2688)
> Cannot find type definition file for 'source-map'. (2688)
> Cannot find type definition file for 'uglify-js'. (2688)
> Cannot find type definition file for 'webpack'. (2688)
>     at getOutput (C:\xyz\node_modules\ts-node\src\index.ts:312:15)
>     at C:\xyz\node_modules\ts-node\src\index.ts:343:16
>     at Object.compile (C:\xyz\node_modules\ts-node\src\index.ts:459:19)
>     at Module.m._compile (C:\xyz\node_modules\ts-node\src\index.ts:395:43)
>     at Module._extensions..js (module.js:579:10)
>     at Object.require.extensions.(anonymous function) [as .js] (C:\xyz\node_modules\ts
> -node\src\index.ts:398:12)
>     at Module.load (module.js:487:32)
>     at tryModuleLoad (module.js:446:12)
>     at Function.Module._load (module.js:438:3)
>     at Module.require (module.js:497:17)
>     at require (internal/module.js:20:19)
>     at Object.parseConfig (C:\xyz\node_modules\karma\lib\config.js:363:22)
>     at new Server (C:\xyz\node_modules\karma\lib\server.js:56:20)
>     at Object.exports.run (C:\xyz\node_modules\karma\lib\cli.js:280:7)
>     at Object.<anonymous> (C:\xyz\node_modules\karma\bin\karma:3:23)
>     at Module._compile (module.js:570:32)

**Expected behavior:**
I would expect that the code correctly compile and run my tests.

**Additional information:**
Version 3.2.0 is the latest version of ts-node available on NPM and the one currently installed when I ran into this issue. I resolved this problem by rolling back to ts-node@3.1.0. I didn't make any other changes to the code, to configuration files or to the installed versions of other packages, but it now successfully builds without error and runs my tests.

**Comment by blakeembrey** [comment]
Feel free to create a reproduction and I'll review it. Otherwise you can review the changes here: https://github.com/TypeStrong/ts-node/commit/3f0d97520ccd7574566e5970cc511937f06599af#diff-f41e9d04a45c83f3b6f6e630f10117feL542. That's the only code that changed how paths are read, but if anything I would have expected an improvement - I'm literally using the `ts.sys` methods now.

**Comment by WhitWaldo** [comment]
I'll see if I can't identify a smaller repro to share and get back to you.

**Comment by cmjartan** [comment]
Same problem here, only that downgrading `ts-node` does not help. This bug only occurs after an update to yarn 0.27.5. With yarn 0.24.6 it's working fine.

**Comment by FrozenPandaz** [comment]
I'm also running into this with karma tests?

Started to occur after upgrading to typescript 2.4.1 but was able to fix it by removing ts-node...

**Comment by rajgopalv** [comment]
1usmvaithiya1:~ vaithiya$ npm install -g ts-node
/usr/local/bin/ts-node -> /usr/local/lib/node_modules/ts-node/dist/bin.js
/usr/local/bin/_ts-node -> /usr/local/lib/node_modules/ts-node/dist/_bin.js
/usr/local/lib
└─┬ ts-node@3.2.0 
  └─┬ chalk@2.0.1
    └─┬ ansi-styles@3.1.0
      └─┬ color-convert@1.9.0
        └── color-name@1.1.3 

1usmvaithiya1:~ vaithiya$ npm install typescript -g
/usr/local/bin/tsc -> /usr/local/lib/node_modules/typescript/bin/tsc
/usr/local/bin/tsserver -> /usr/local/lib/node_modules/typescript/bin/tsserver
/usr/local/lib
└── typescript@2.4.1 

1usmvaithiya1:~ vaithiya$ ts-node 
> console.log('hello');
⨯ Unable to compile TypeScript
[eval].ts: Cannot find name 'exports'. (2304)
[eval].ts (0,11): Cannot find name 'module'. (2304)
>  
(To exit, press ^C again or type .exit)
> 
1usmvaithiya1:~ vaithiya$

**Comment by blakeembrey** [comment]
@rajgopalv That issue is unrelated, but logged - see https://github.com/TypeStrong/ts-node/issues/351.

**Comment by iDaN5x** [comment]
**Experiencing the same issue with:**
ts-node v3.3.0
node v8.4.0
typescript v2.4.2

**Reproduction**
```typescript
~  ts-node
> const x = 5;
Thrown: ⨯ Unable to compile TypeScript
[eval].ts: Cannot find name 'exports'. (2304)
[eval].ts (0,11): Cannot find name 'module'. (2304)
```

**Comment by SamuelMarks** [comment]
Same problem here:
```
node@8.4.0
ts-node@3.3.0
typescript@2.5.2
```

**Comment by Olian04** [comment]
Having the same issue as @iDaN5x 

> ts-node v3.3.0
> node v8.5.0
> typescript v2.5.2

**Comment by winksaville** [comment]
@Olian04 @SamuelMarks @iDaN5x,

Assuming all of you were trying to run `ts-node` in REPL mode the problem can be worked around by installing @types/node in a working directory, such as ts-node-repl, and then `npm install @types/node`. And then use `ts-node` in that directory as @blakeembrey mentioned in [issue 351](https://github.com/TypeStrong/ts-node/issues/351#issuecomment-307179726). Apparently this will be fixed "properly" in ts-node [release 4.0](https://github.com/TypeStrong/ts-node/issues/351#issuecomment-329844650)

**Comment by blakeembrey** [comment]
Closing as it should be resolved since 4.0. Thanks @winksaville.
